Source/WebCore/platform/graphics/MediaPlayer.cpp

```diff
@@ -1392,6 +1396,16 @@ bool MediaPlayer::supportsKeySystem(const String& keySystem, const String& mimeT
     return false;
 }
 
+bool MediaPlayer::hasThreadSafeSupport()
+{
+#if USE(AVFOUNDATION)
+    Locker locker { s_lock };
+    return !!registerRemotePlayerCallback();
```

Comment:
It is a bit weird to call the callback in a bool getter.

Reply:
At this stage, it is the only data available in WebCore that identify media in the GPU process being used. And it is the presence of that callback that is used throughout MediaPlayer and other code to determine if we are to use the GPU process.

so a bit weird yes, but existing, I only added the existence of the callback.

Reply:
ok, the callback is set if the pref `UseGPUProcessForMediaEnabled` is set, and where the callback is set is done just after setting the m_useGPUProcess on the MediaStrategy, so it's equivalent; I've removed the code entirely.